src/jl_uv.c

```diff
@@ -333,6 +375,7 @@ JL_DLLEXPORT int jl_fs_unlink(char *path)
 {
     uv_fs_t req;
     JL_SIGATOMIC_BEGIN();
+    // no callback, no lock needed TODO: remove any blocking calls
```

Comment:
Are you sure about this? (and all the other comment about callbacks). Is `libuv` itself thread safe now or does it have a list of thread safe functions even when they share the same event loop?

Reply:
I inspected the libuv implementation and in all these calls, it should be safe to pass NULL instead of the loop argument since when the callback is NULL, it does everything directly using blocking calls to OS and locally allocated memory. All the loop-global async structures are just bypassed.

Reply:
Are those guaranteed/documented?

Reply:
Do you prefer sitting in a lock blocking any further progress on other threads which want to do IO? If it breaks, the right fix will be as described in TODO - replace the blocking call with the asynchronous one, then the lock will not affect performance

Reply:
It depends.... I just want to know if it's actually guaranteed or not. Having a TODO with a easy hot fix if needed is fine. On a case by case basis we could also get rid of the libuv wrapper for some of these, similar to what we did with signals.